### PR TITLE
clipper2: add version 1.2.3

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **clipper2/1.2.3**

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/clipper2/all/conanfile.py [hunk 1]
```diff
@@ -66,7 +66,7 @@ def validate(self):
             )
 
     def source(self):
-        get(self, **self.conan_data["sources"][self.version])
+        get(self, **self.conan_data["sources"][self.version], strip_root = Version(self.version) >= "1.2.3")
```
Comment (RubenRBS) [inline recipes/clipper2/all/conanfile.py:69]: I'm curious, what changed in the new version for this to be needed?

Comment (toge) [reply recipes/clipper2/all/conanfile.py:69]: @RubenRBS 
I am asking about the release file on this issue. https://github.com/AngusJohnson/Clipper2/issues/670

**Review by jwillikers — APPROVED** [review]

**Review by jwillikers — APPROVED** [review]

**Comment by RubenRBS** [comment]
Seems to be building now, will relaunch monday if it does not get there by then

File: recipes/clipper2/all/conanfile.py (hunk 1 above)
Comment (RubenRBS) [reply recipes/clipper2/all/conanfile.py:69]: Thanks! Their response confuses me a little, is this then expected?

**Review by jwillikers — APPROVED** [review]

**Pull request merged**